src/VersionedStateExtension.php

```diff
@@ -11,7 +11,8 @@
 /**
  * Persists versioned state between requests via querystring arguments
  *
- * @property RequestHandler|DataObject $owner
+ * @template T of RequestHandler|DataObject
+ * @extends Extension<T>
```

Comment:
```suggestion
 * @extends Extension<RequestHandler|DataObject>
```
Wouldn't this do the same thing?

Reply:
Nope. Having the `@template T` there means subclasses can say "Use my type instead" using `@extends`.
`VersionedTestSessionExtension` is a subclass of `VersionedStateExtension` and has its own type, so `VersionedStateExtension` has to use `@template T` to allow that to happen.